### DISCUSSION TITLE
Mastering Code Searches: Community Tips and Tricks for Pro-Level Code Discovery

### DISCUSSION THREAD

**Discussion by LiteBrite82** [community/community#138837] [Other Feature Feedback, Questions, & Ideas]
# Mastering Code Searches: Community Tips and Tricks for Pro-Level Code Discovery
**Ever wish you could find the exact piece of code you need in record time?**

**Are you Struggling with complex queries and not getting the results you want?**

You’re in the right place! This month, let’s join forces to create a go-to resource for locating hard-to-find code on GitHub.
We’re focusing on searching from the GitHub search UI rather than using the API.

**What We’re Looking For:**
- **Effective Search Strings:** Share the queries that make your workflow smoother.
- **Overcoming Search Challenges:** Tell us about obstacles you’ve faced and how you tackled them.
- **Pro Tips and Tricks:** Any clever techniques or insights you’ve discovered for better code searching.

Post your best strategies and experiences here, and we’ll compile them into a single, pinned discussion for everyone to benefit from. Let’s elevate our code searching game together!

**Comment by XDYB** [comment]
I would like to ask, can code search be sorted by time? Otherwise, this discussion is meaningless, and the code from 10 years ago is useless no matter how accurate it is.
我想问一下，代码搜索可以按时间排序吗？否则，这个讨论就没有意义，10年前的代码再准确也没用。

**Reply by LiteBrite82** [reply]
Code cannot be sorted by date in the code search UI unfortunately. Code is time stamped by the date it was committed, so you need to be searching for commits that contain that code. 

Unfortunately, you can only search a commit by strings in the commit message. If a developer includes detailed commit messages that include code snippets, that is a possible avenue. 

If you have a specific repo you are searching, and can pull it to your local machine, you can use the CLI to search for it. You could also use GitHub Codespaces to the same effect.

**Comment by LiteBrite82** [comment]
Here is a helpful tip from @ClouddJR 's blog [Making the most of GitHub Code Search](https://arkadiuszchmura.com/posts/making-the-most-of-github-code-search/)

> Definition of a class
The `symbol:` qualifier is one of the most useful ones. You could use it to find the definition of a specific class (or any other symbol, like a function) when you don’t know the repository the class is in. The query could look like this: `symbol:HelloController org:github language:Ruby`.

>This way of searching is especially handy when your company uses a microservices architecture and the code is distributed across hundreds (or thousands) of repositories.

**Comment by llvee** [comment]
Thank you @LiteBrite82 for opening this discussion. Current search functionality does not provide enough control. I am interested in helping engineer better code search functionality.

**Comment by Hanoi1989** [comment]
> Thank you @LiteBrite82 for opening this discussion. Current search functionality does not provide enough control. I am interested in helping engineer better code search functionality.
